### ISSUE TITLE
Proxy Server Customization

### ISSUE THREAD

**Issue by jasoncouture** [braintree/braintree_dotnet#31]
Proxy server does not allow credentials to be specified.
Please expose IWebProxy property instead, allowing something along the lines of:

```
client.Configuration.WebProxy = new WebProxy("http://some.proxy.server:3128/", false, null, new NetworkCredential(user, password));
```

or in our case:
```
client.Configuration.WebProxy = DependencyInjector.Resolve<IProxyFactory>().GetProxy();
```

**Comment by pblesi** [comment]
@jasoncouture 

Thanks for bring this to our attention. We will look to add this functionality in an upcoming release and will notify you once this feature is available.

**Comment by jackellenberger** [comment]
@jasoncouture

Thanks again for pointing this out, we are currently finalizing changes to all of our client libraries to add authenticated proxy functionality. Look for it in the next release.

**Comment by jasoncouture** [comment]
@jackellenberger Thanks for the update. I was actually considering adding a pull request, but now i'll wait. haha.

**Comment by omgrr** [comment]
@jasoncouture We just released 2.64.0 which contains these changes. Let us know if you run into any problems!
